crates/ark/src/lsp/symbols.rs

```diff
@@ -414,26 +532,41 @@ fn collect_call_arguments(
                 return Ok(());
             };
 
-            // Recurse into arguments. They might be a braced list, another call
-            // that might contain functions, etc.
-            collect_symbols(ctx, &arg_value, contents, 0, symbols)?;
-
+            // If this is a named function, collect it as a method (new node in the tree)
             if arg_value.kind() == "function_definition" {
                 if let Some(arg_fun) = child.child_by_field_name("name") {
-                    // If this is a named function, collect it as a method
                     collect_method(ctx, &arg_fun, &arg_value, contents, symbols)?;
-                } else {
-                    // Otherwise, just recurse into the function
-                    let body = arg_value.child_by_field_name("body").into_result()?;
-                    collect_symbols(ctx, &body, contents, 0, symbols)?;
+                    return Ok(());
                 };
+                // else fallthrough
             }
 
+            collect_symbols(ctx, &arg_value, contents, 0, symbols)?;
+
             Ok(())
         },
     )
 }
 
+fn collect_function_parameters(
+    ctx: &mut CollectContext,
+    node: &Node,
+    contents: &Rope,
+    symbols: &mut Vec<DocumentSymbol>,
+) -> anyhow::Result<()> {
+    collect_sections(
+        ctx,
+        &node,
+        contents,
+        0,
+        symbols,
+        |_ctx, _child, _contents, _symbols| {
+            // Only collect sections
+            return Ok(());
```

Comment:
I'm not sure what this comment means. Does it mean no recursion? Can you provide more details?

Reply:
yep no recursion, the comment is now clearer about that